plugin/trino-opensearch/src/main/java/io/trino/plugin/opensearch/OpenSearchMetadata.java

```diff
@@ -672,6 +656,20 @@ public Optional<TableFunctionApplicationResult<ConnectorTableHandle>> applyTable
         return Optional.of(new TableFunctionApplicationResult<>(tableHandle, columnHandles));
     }
 
+    private static boolean supportsPredicates(Type type)
+    {
+        return switch (type) {
+            case TimestampType ignore -> true;
+            case BooleanType ignore -> true;
+            case TinyintType ignore -> true;
+            case SmallintType ignore -> true;
+            case IntegerType ignore -> true;
+            case BigintType ignore -> true;
+            case RealType ignore -> true;
+            default -> false;
```

Comment:
Return false for `DoubleType` as `scaled_float` and `double` maps to `DoubleType` and `scaled_float` isn't pushed down in removed method.
Return false for `VarcharType` as `text` and `keyword` types maps to `VarcharType` and `text` isn't pushed down in removed method.